### PR TITLE
CSW-53&67 - Improve text for S3 ACL and RDS classic security group checks

### PR DESCRIPTION
**Pull request by danjoneslf**
I've made a few changes here: 
1. The trusted advisor check get_data method now does a looped retry to refresh the check status to ensure we have current data. I've added a max retry count so it doesn't keep refreshing forever. It's also has a non-actioned except because some TA checks raise a "not refreshable" exception.
2. Add a get_resource_data method to the TA checks to get the raw data associated with the failing resources - This is overridden per check so it doesn't fail if not implemented in a given check.
3. Change the annotation and how_do_i_fix_it entries for the 2 checks mentioned.

### CODE REVIEW

File: chalice/chalicelib/aws/gds_ec2_security_group_client.py [hunk 1]
```diff
@@ -23,6 +23,23 @@ def describe_security_groups(self, session, **kwargs):
...
+
+            # run describe security groups api call
+            response = ec2.describe_security_groups(
+                GroupIds = [id]
+            )
+
+            group = None
+            if len(response["SecurityGroups"] == 1):
+                group = response["SecurityGroups"][0]
+        except Exception as err:
```
Comment (0atman) [inline chalice/chalicelib/aws/gds_ec2_security_group_client.py]: Can we interrogate this exception?

File: chalice/chalicelib/aws/gds_s3_client.py [hunk 2]
```diff
@@ -40,10 +40,10 @@ def get_bucket_versioning(self, session, bucket_name):
         try:
             self.app.log.debug(f"Get bucket versioning for ({bucket_name})")
             # Don't try to get versioning without a name
-            if bucket_name is not None and bucket_name != "":
+            if bucket_name:
```
Comment (0atman) [inline chalice/chalicelib/aws/gds_s3_client.py]: :fire:

File: chalice/chalicelib/aws/gds_ec2_security_group_client.py (hunk 1 above)
Comment (danjoneslf) [reply chalice/chalicelib/aws/gds_ec2_security_group_client.py]: I've pushed a change to raise an exception if id is None and error log any exceptions raised.

Comment (0atman) [reply chalice/chalicelib/aws/gds_ec2_security_group_client.py]: LGTM :+

**Pull request merged**